### PR TITLE
[Feature] Extract iteration result based on the payload found in output instead of requester

### PR DESCRIPTION
**Pull request by Bobowski**
# Description

In order to find vulnerabilities that were triggered by one scenario but found in other requests' response we need to change the way we extract iteration results while building vulnerabilities list. The change is just to check if the payload_guild found in vulnerability is actually coming from the execution that it's linked to, it they are then it's just a standard case for "this request caused this vulnerability" in other case we need to extract different iteration_result (based on the payload_guid because altho we found the vulnerability in this output it was not caused by this request.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Yes, it's simple logical change. 

# Checklist:

- [x] PR title contains the nature of changes (Docs, Feature, Fix, CI, etc.)
- [x] I've set corresponding reviewers, set assignees, and labels.
- [x] My code follows the style guidelines of this project (running `pre-commit`)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by jjakiteli — APPROVED** [review]

**Review by dawid-czarnecki — CHANGES_REQUESTED** [review]
This has to be tested once annotations are properly removed. Here is a quick patch on cerebrate that introduce stored XSS. Injection in `/AuthKeys/add`, exploitation in `/AuditLogs/index`:
```
diff --git a/src/Controller/AuditLogsController.php b/src/Controller/AuditLogsController.php
index e3e4e0ce..85709159 100644
--- a/src/Controller/AuditLogsController.php
+++ b/src/Controller/AuditLogsController.php
@@ -17,6 +17,17 @@ class AuditLogsController extends AppController
 
     public function index()
     {
+        $keys = $this->getTableLocator()->get('AuthKeys');
+        $all = $keys->find()->all();
+        if($all->count()>0) {
+            $all = $all->toArray();
+            foreach($all as $key) {
+                $id = $key->get('id');
+                $comment = $key->get('comment');
+                print("Key: $id. Comment: $comment<br>\n");
+            }
+            die;
+        }
         $this->CRUD->index([
             'contain' => $this->containFields,
             'order' => ['AuditLogs.id' => 'DESC'],
```

File: cakefuzzer/domain/components.py [hunk 1]
```diff
@@ -407,24 +418,65 @@ async def save_to_file(self, filename: Path = Path("results.json")):
...
-                        "_REQUEST": vuln.iteration_result.output.REQUEST,
-                        "_COOKIE": vuln.iteration_result.output.COOKIE,
-                        "_FILES": vuln.iteration_result.output.FILES,
-                        "_SERVER": vuln.iteration_result.output.SERVER,
-                    },
-                }
+
+                if vuln.iteration_result == vuln.found_in_iteration_result:
+                    dump_vuln = {
+                        "strategy_name": vuln.iteration_result.scenario.strategy_name,
```
Comment (Bobowski) [inline cakefuzzer/domain/components.py:424]: add "found_in" as to have a common dict structure

File: cakefuzzer/domain/components.py [hunk 2]
```diff
@@ -414,24 +418,65 @@ async def save_to_file(self, filename: Path = Path("results.json")):
...
+                            "_SERVER": vuln.iteration_result.output.SERVER,
+                        },
+                    }
+                else:
+                    dump_vuln = {
+                        "triggered_by":{
+                            "strategy_name": vuln.iteration_result.scenario.strategy_name,
+                            "payload": payload,
+                            "detection_result": vuln.vulnerability.detection_result,
+                            "context_location": vuln.vulnerability.context_location,
```
Comment (dawid-czarnecki) [inline cakefuzzer/domain/components.py:447]: These can be removed

File: cakefuzzer/domain/components.py [hunk 3]
```diff
@@ -414,24 +418,65 @@ async def save_to_file(self, filename: Path = Path("results.json")):
...
+                                "_POST": vuln.iteration_result.output.POST,
+                                "_REQUEST": vuln.iteration_result.output.REQUEST,
+                                "_COOKIE": vuln.iteration_result.output.COOKIE,
+                                "_FILES": vuln.iteration_result.output.FILES,
+                                "_SERVER": vuln.iteration_result.output.SERVER,
+                            },
+                        },
+                        "found_in":{
+                            "strategy_name": vuln.found_in_iteration_result.scenario.strategy_name,
+                            "payload": vuln.found_in_iteration_result.scenario.payload,
```
Comment (dawid-czarnecki) [inline cakefuzzer/domain/components.py:463]: This can be removed.

File: cakefuzzer/domain/components.py [hunk 4]
```diff
@@ -414,24 +418,65 @@ async def save_to_file(self, filename: Path = Path("results.json")):
...
+                            "detection_result": vuln.vulnerability.detection_result,
+                            "context_location": vuln.vulnerability.context_location,
+                            "vulnerability_location": vuln.vulnerability_location,
+                            "vulnerability_id": i,
```
Comment (dawid-czarnecki) [inline cakefuzzer/domain/components.py:467]: This can be removed.

**Review by dawid-czarnecki — CHANGES_REQUESTED** [review]
Other than that the PR is ready to merge. No need to change `results.json` more at this point.

**Review by dawid-czarnecki — APPROVED** [review]
LGTM

**Pull request merged**